Dataset.py

```diff
@@ -377,6 +396,9 @@ def get_seq_order_for_epoch(self, epoch, num_seqs, get_seq_len=None):
     This is mostly a static method, except that is depends on the configured type of ordering,
     such as 'default' (= as-is), 'sorted' or 'random'. 'sorted' also uses the sequence length.
 
+    TODO: This function as to be used in the subclasses explicitely, so it is possible to write a dataset
+      that does not support the seq_ordering parameter and it will never cause a warning or assert
```

Comment:
It is possible to write all sorts of buggy or strange code. You cannot avoid that.

A good behaving subclass should of course check for `seq_ordering` (in `init_seq_order`), and do the corresponding thing, and if it cannot handle some case, it should throw an exception. I think most datasets do this correctly, or not?